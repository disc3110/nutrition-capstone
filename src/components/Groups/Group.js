import { NavLink } from 'react-router-dom';

const Group = (props) => {
  const { name, number, slug } = props

  return (
    <NavLink className="links p-3 min-h-5 group" to={`/sports/${slug}`} >
        <h2 className="text-end no-select fw-bold">{name.toUpperCase()}</h2>
        <p className="text-end no-select">{`${number} sports`}</p>
    </NavLink>
  )
}

export default Group