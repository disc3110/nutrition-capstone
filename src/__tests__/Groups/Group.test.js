import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import {
  HashRouter as Router,
} from 'react-router-dom';
import Group from '../../components/Groups/Group';
import store from '../../Redux/configureStore';

describe('Check if all components are rendered', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Router><Group /></Router></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
