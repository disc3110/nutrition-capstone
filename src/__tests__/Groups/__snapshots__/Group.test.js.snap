// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if all components are rendered renders correctly 1`] = `
<a
  aria-current={null}
  className="links p-3 min-h-5 group"
  href="#/sports/undefined"
  onClick={[Function]}
>
  <h2
    className="text-end no-select fw-bold"
  >
    
  </h2>
  <p
    className="text-end no-select"
  >
    undefined sports
  </p>
</a>
`;
